hamilton/plugins/polars_extensions.py

```diff
@@ -499,6 +507,146 @@ def name(cls) -> str:
         return "json"
 
 
+@dataclasses.dataclass
+class PolarsSpreadsheetReader(DataLoader):
+    """
+    Class specifically to handle loading Spreadsheet files with Polars.
+    Should map to https://pola-rs.github.io/polars/py-polars/html/reference/api/polars.read_excel.html
+    """
+
+    source: Union[str, BytesIO, Path, BinaryIO, bytes]
+    sheet_id = None
+    sheet_name = None
+    engine: Literal['xlsx2csv', 'openpyxl', 'pyxlsb'] = None
+    xlsx2csv_options: Dict[str, Any] = None
+    read_csv_options: Dict[str, Any] = None
+    schema_overrides: SchemaDict = None
+    raise_if_empty: bool = True
+
+    @classmethod
+    def applicable_types(cls) -> Collection[Type]:
+        return [DATAFRAME_TYPE]
+
+    def _get_loading_kwargs(self):
+        kwargs = {}
+        if self.sheet_id is not None:
+            kwargs["sheet_id"] = self.sheet_id
+        if self.sheet_name is not None:
+            kwargs["sheet_name"] = self.sheet_name
+        if self.engine is not None:
+            kwargs["engine"] = self.engine
+        if self.xlsx2csv_options is not None:
+            kwargs["xlsx2csv_options"] = self.xlsx2csv_options
+        if self.read_csv_options is not None:
+            kwargs["read_csv_options"] = self.read_csv_options
+        if self.schema_overrides is not None:
+            kwargs["schema_overrides"] = self.schema_overrides
+        if self.raise_if_empty is not None:
+            kwargs["raise_if_empty"] = self.raise_if_empty
+        return kwargs
+
+    def load_data(self, type_: Type) -> Tuple[DATAFRAME_TYPE, Dict[str, Any]]:
+        df = pl.read_excel(self.source, **self._get_loading_kwargs())
+        metadata = utils.get_file_metadata(self.source)
+        return df, metadata
+
+    @classmethod
+    def name(cls) -> str:
+        return "excel"
+
+
+@dataclasses.dataclass
+class PolarsSpreadsheetWriter(DataSaver):
+    """
+    Class specifically to handle saving Spreadsheet files with Polars.
+    Should map to https://pola-rs.github.io/polars/py-polars/html/reference/api/polars.DataFrame.write_excel.html
+    """
+
+    workbook: Union[Workbook, BytesIO, str, Path] = None
```

Comment:
I think we shouldn't allow this to be None. 
I don't think the default that polars has makes sense here. So I think we want someone to always input something here to direct where to save it.